### PR TITLE
Fix installation/uninstallation of notifications for GLPI 9.2, add so…

### PR DESCRIPTION
**Pull request by btry**
…me tests

### CODE REVIEW

File: install/install.php [hunk 1]
```diff
@@ -299,35 +311,39 @@ protected function createNotifications() {
...
-      $this->migration->displayMessage("Delete notifications");
-
       // Define DB tables
-      $table_targets      = getTableForItemType('NotificationTarget');
-      $table_notification = getTableForItemType('Notification');
-      $table_translations = getTableForItemType('NotificationTemplateTranslation');
-      $table_templates    = getTableForItemType('NotificationTemplate');
+      $table_targets       = NotificationTarget::getTable();
+      $table_notification  = Notification::getTable();
+      $table_translations  = NotificationTemplateTranslation::getTable();
```
Comment (orthagh) [inline install/install.php]: ``$table_translations`` not used anymore

File: install/install.php [hunk 2]
```diff
@@ -299,35 +311,39 @@ protected function createNotifications() {
...
+      $table_templates     = NotificationTemplate::getTable();
 
       // Delete translations
       $query = "DELETE FROM `$table_translations`
-      WHERE `notificationtemplates_id` IN (
-      SELECT `id` FROM $table_templates WHERE `itemtype` = 'PluginFormcreatorForm_Answer')";
+                WHERE `notificationtemplates_id` IN (
```
Comment (orthagh) [inline install/install.php]: like below, could you also replace this direct sql by a ``deleteByCriteria`` (and so remove the getTable above)

File: install/install.php [hunk 3]
```diff
@@ -299,35 +311,39 @@ protected function createNotifications() {
...
-      WHERE `itemtype` = 'PluginFormcreatorForm_Answer'";
-      $DB->query($query);
+      $template = new NotificationTemplate();
+      $template->deleteByCriteria(['itemtype' => 'PluginFormcreatorForm_Answer']);
 
       // Delete notification targets
       $query = "DELETE FROM `$table_targets`
-      WHERE `notifications_id` IN (
-      SELECT `id` FROM $table_notification WHERE `itemtype` = 'PluginFormcreatorForm_Answer')";
+                WHERE `notifications_id` IN (
```
Comment (orthagh) [inline install/install.php]: same (``deleteByCriteria``)

**Comment by btry** [comment]
This PR needs a patch for GLPI 

https://github.com/glpi-project/glpi/pull/2736

**Pull request merged**